library/core/src/marker.rs

```diff
@@ -668,24 +668,17 @@ impl<T: ?Sized> !Sync for *mut T {}
 ///
 /// ## Ownership and the drop check
 ///
-/// Adding a field of type `PhantomData<T>` indicates that your
-/// type owns data of type `T`. This in turn implies that when your
-/// type is dropped, it may drop one or more instances of the type
-/// `T`. This has bearing on the Rust compiler's [drop check]
-/// analysis.
-///
-/// If your struct does not in fact *own* the data of type `T`, it is
-/// better to use a reference type, like `PhantomData<&'a T>`
-/// (ideally) or `PhantomData<*const T>` (if no lifetime applies), so
-/// as not to indicate ownership.
+/// Adding a field of type `PhantomData<T>` indicates that your type *owns* data of type `T`. This
+/// in turn has effects on the Rust compiler's [drop check] analysis, but that only matters in very
+/// specific circumstances. For the exact rules, see the [drop check] documentation.
```

Comment:
```suggestion
/// The exact interaction of `PhantomData` with drop check may change in the future.
///
/// Adding a field of type `PhantomData<T>` currently indicates that your type *owns* data of type
/// `T` in very rare circumstances. This in turn has effects on the Rust compiler's [drop check]
/// analysis. For the exact rules, see the [drop check] documentation.
```

Reply:
I don't understand why you recommended this change, @lcnr. It seems to obscure the matter more from the user perspective: does PhantomData indicate you own its type parameter or not?

Reply:
well, currently it does, in the future it likely will stop doing so